benchmark/Streamly/Benchmark/Data/Fold.hs

```diff
@@ -57,174 +55,151 @@ moduleName = "Data.Fold"
-        [ bgroup "composition" -- Applicative
-            [ benchIOSink
-                  value
-                  "all,any"
-                  (S.fold
-                       ((,) <$> FL.all (<= (value + 1)) <*>
-                        FL.any (> (value + 1))))
-            , benchIOSink
-                  value
-                  "sum,length"
-                  (S.fold ((,) <$> FL.sum <*> FL.length))
-            , benchIOSink
-                  value
-                  "tee sum length"
-                  (S.fold (FL.tee FL.sum FL.length))
-            , benchIOSink
-                  value
-                  "distribute sum length"
-                  (S.fold (FL.distribute [FL.sum, FL.length]))
-            , benchIOSink
-                  value
-                  "partition sum length"
-                  (S.fold (IFL.lmap
-                               (\a -> if odd a then Left a else Right a)
-                               (FL.partition FL.sum FL.length)))
-            , benchIOSink
-                  value
-                  "unzip sum length"
-                  (S.fold (IFL.lmap (\a -> (a, a)) (FL.unzip FL.sum FL.length)))
-            ]
+    [ bgroup "composition" -- Applicative
+        [ benchIOSink
+              value
+              "all,any"
+              (S.fold
+                   ((,) <$> FL.all (<= (value + 1)) <*>
+                    FL.any (> (value + 1))))
+        , benchIOSink
+              value
+              "sum,length"
+              (S.fold ((,) <$> FL.sum <*> FL.length))
+        , benchIOSink
+              value
+              "tee sum length"
+              (S.fold (FL.tee FL.sum FL.length))
+        , benchIOSink
+              value
+              "distribute sum length"
+              (S.fold (FL.distribute [FL.sum, FL.length]))
+        , benchIOSink
+              value
+              "partition sum length"
+              (S.fold (IFL.lmap
+                           (\a -> if odd a then Left a else Right a)
+                           (FL.partition FL.sum FL.length)))
+        , benchIOSink
+              value
+              "unzip sum length"
+              (S.fold (IFL.lmap (\a -> (a, a)) (FL.unzip FL.sum FL.length)))
         ]
     ]
 
-o_n_heap_serial :: Int -> [Benchmark]
-o_n_heap_serial value =
-    [ bgroup "serially"
-          [ bgroup "elimination"
-          -- Left folds for building a structure are inherently non-streaming
-          -- as the structure cannot be lazily consumed until fully built.
-                [ benchIOSink value "toStream" (S.fold IP.toStream)
-                , benchIOSink value "toStreamRev" (S.fold IP.toStreamRev)
-                , benchIOSink value "toList" (S.fold FL.toList)
-                , benchIOSink value "toListRevF" (S.fold IFL.toListRevF)
-          -- Converting the stream to an array
-                , benchIOSink value "lastN.Max" (S.fold (IA.lastN (value + 1)))
-                , benchIOSink value "writeN" (S.fold (A.writeN value))
```

Comment:
These two benchmarks can go to the Array benchmarks, as they belong to array modules.